packages/host/app/resources/adoption-chain-manager.ts

```diff
@@ -0,0 +1,103 @@
+import { ImportResource } from './import';
+import { Resource } from 'ember-resources';
+import { tracked } from '@glimmer/tracking';
+import { Loader } from '@cardstack/runtime-common/loader';
+import { getOwner } from '@ember/application';
+import { type BaseDef } from 'https://cardstack.com/base/card-api';
+import type LoaderService from '../services/loader-service';
+import {
+  getCardType,
+  type CardType,
+} from '@cardstack/host/resources/card-type';
+import { restartableTask } from 'ember-concurrency';
+import { action } from '@ember/object';
+
+interface AdoptionChainManagerArgs {
+  named: { importResource: ImportResource | undefined; loader: Loader };
+}
+
+export type ElementInFile = CardInFile; // can add more types here
+
+interface CardInFile {
+  cardType: CardType;
+  card: typeof BaseDef;
+}
+
+// forwards the card type resource (doesn't care about the card type resource)
+export class AdoptionChainManager extends Resource<AdoptionChainManagerArgs> {
+  importResource: ImportResource | undefined;
+  ready: Promise<void> | undefined;
+  @tracked elementsInFile: ElementInFile[] = [];
+  @tracked selected: ElementInFile | undefined;
```

Comment:
is this public API for this resource? you have a selected action that sets this field so i'm guessing this should be private right?